### PR TITLE
[freetype] Add cmake policy CMP0011 and CMP0012

### LINKED ISSUE #17055: [nuklear] fatal error: 'malloc.h' file not found
**Is your feature request related to a problem? Please describe.**

### `CMakeLists.txt`
```cmake
cmake_minimum_required(VERSION 3.0)
cmake_policy(SET CMP0048 NEW)
project(pp VERSION 0.0.0 LANGUAGES C)

set(src src/main.c)

find_path(NUKLEAR_INCLUDE_DIRS "nuklear.h")

add_executable(${PROJECT_NAME} ${src})
target_include_directories(${PROJECT_NAME} PRIVATE ${NUKLEAR_INCLUDE_DIRS})
````

### `src/main.c`
```c
#include <nuklear.h>
#include <malloc.h>
#include <allegro5/allegro.h>
#define MAX_MEMORY 1024

int main(void) {
    /* init gui state */
    struct nk_context ctx;
    NkAllegro5Font *font;
    font = nk_allegro5_font_create_from_file("../../../extra_font/Roboto-Regular.ttf", 12, 0);
    nk_init_fixed(&ctx, calloc(1, MAX_MEMORY), MAX_MEMORY, &font);

    enum {EASY, HARD};
    static int op = EASY;
    static float value = 0.6f;
    static int i =  20;

    if (nk_begin(&ctx, "Show", nk_rect(50, 50, 220, 220),
                 NK_WINDOW_BORDER|NK_WINDOW_MOVABLE|NK_WINDOW_CLOSABLE)) {
        /* fixed widget pixel width */
        nk_layout_row_static(&ctx, 30, 80, 1);
        if (nk_button_label(&ctx, "button")) {
            /* event handling */
        }

        /* fixed widget window ratio width */
        nk_layout_row_dynamic(&ctx, 30, 2);
        if (nk_option_label(&ctx, "easy", op == EASY)) op = EASY;
        if (nk_option_label(&ctx, "hard", op == HARD)) op = HARD;

        /* custom widget pixel width */
        nk_layout_row_begin(&ctx, NK_STATIC, 30, 2);
        {
            nk_layout_row_push(&ctx, 50);
            nk_label(&ctx, "Volume:", NK_TEXT_LEFT);
            nk_layout_row_push(&ctx, 110);
            nk_slider_float(&ctx, 0, &value, 1.0f, 0.1f);
        }
        nk_layout_row_end(&ctx);
    }
    nk_end(&ctx);
}
```

### `shell`
```sh
$ [path]/vcpkg install nuklear
$ mkdir build && cd $_
$ cmake -DCMAKE_TOOLCHAIN_FILE=~/repos/vcpkg/scripts/buildsystems/vcpkg.cmake ..
$ cmake --build .
/tmp/b/src/main.c:2:10: fatal error: 'malloc.h' file not found
#include <malloc.h>
         ^~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/pp.dir/src/main.c.o] Error 1
make[1]: *** [CMakeFiles/pp.dir/all] Error 2
make: *** [all] Error 2
```

**Proposed solution**
Would be nice if the example would build

**Describe alternatives you've considered**
```c
#if defined(macintosh) || defined(Macintosh) || defined(__APPLE__) && defined(__MACH__)
#include <stdlib.h>
#else
#include <malloc.h>
#endif
````
(but would need to be more careful, there is a list of macros on sourceforge lined up with where to find malloc, so I bunch of these checks are required)

But it could find neither `#include "nuklear.h"` nor `#include <nuklear.h>`…

### PR DESCRIPTION
**Pull request by JackBoosY**
Fix bug when using freetype:
```
CMake Warning (dev) at /home/work/vcpkg/installed/x64-linux/share/freetype/vcpkg-cmake-wrapper.cmake:3 (if):
  if given arguments:

    "ON"

  An argument named "ON" appears in a conditional statement.  Policy CMP0012
  is not set: if() recognizes numbers and boolean constants.  Run "cmake
  --help-policy CMP0012" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.
Call Stack (most recent call first):
  /home/work/vcpkg/scripts/buildsystems/vcpkg.cmake:736 (include)
  /home/work/vcpkg/downloads/tools/cmake-3.20.2-linux/cmake-3.20.2-linux-x86_64/share/cmake-3.20/Modules/FindX11.cmake:255 (find_package)
  /home/work/vcpkg/scripts/buildsystems/vcpkg.cmake:782 (_find_package)
  CMakeLists.txt:699 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.
```
And
```
CMake Warning (dev) in /home/work/vcpkg/installed/x64-linux/share/freetype/vcpkg-cmake-wrapper.cmake:
  Policy CMP0011 is not set: Included scripts do automatic cmake_policy PUSH
  and POP.  Run "cmake --help-policy CMP0011" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

  The included script

    /home/work/vcpkg/installed/x64-linux/share/freetype/vcpkg-cmake-wrapper.cmake

  affects policy settings.  CMake is implying the NO_POLICY_SCOPE option for
  compatibility, so the effects are applied to the including context.
```

Related: https://github.com/microsoft/vcpkg/issues/17055

### CODE REVIEW

File: ports/freetype/vcpkg-cmake-wrapper.cmake [hunk 1]
```diff
@@ -1,3 +1,6 @@
+cmake_policy(SET CMP0012 NEW)
+cmake_policy(SET CMP0011 NEW)
```
[the remark this answers is not in the document]
Comment (JackBoosY) [reply ports/freetype/vcpkg-cmake-wrapper.cmake]: Please review again.

File: versions/f-/freetype.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "a03c381a612922ba27cf170da915c53c41f9ad30",
```
Comment (JackBoosY) [inline versions/f-/freetype.json:4]:
```suggestion
      "git-tree": "42566110002caeba02e4dc508249bce0aa32a848",
```

File: ports/freetype/vcpkg-cmake-wrapper.cmake [hunk 3]
```diff
@@ -1,65 +1,72 @@
-_find_package(${ARGS})
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.3)
```
Comment (Neumann-A) [inline ports/freetype/vcpkg-cmake-wrapper.cmake:1]: Wait what? `This policy was introduced in CMake version 2.8.0`

Why do you `if` that ? Also you probably only wanted to `if` the `cmake_policy` calls?
(furthermore it looks like a `cmake_minimum_required` call is missing in the calling project? )

`CMAKE_VERSION VERSION_GREATER_EQUAL 3.3`
I also don't get why you use this. I thought we all agreed indirectly to assume at least 3.5.1

Comment (JackBoosY) [reply ports/freetype/vcpkg-cmake-wrapper.cmake:1]: Sorry, I saw many examples on our cmake wrappers before.

Comment (JackBoosY) [reply ports/freetype/vcpkg-cmake-wrapper.cmake]: @Neumann-A Does that met you requirement?

File: ports/freetype/vcpkg-cmake-wrapper.cmake [hunk 4]
```diff
@@ -1,3 +1,8 @@
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.5.1)
```
Comment (dg0yt) [inline ports/freetype/vcpkg-cmake-wrapper.cmake:1]: No `if` needed. `CMP0012` is available in all versions of CMake which vcpkg supports.

File: ports/freetype/vcpkg-cmake-wrapper.cmake [hunk 5]
```diff
@@ -63,3 +68,7 @@ if(FREETYPE_LIBRARIES)
         list(APPEND FREETYPE_LIBRARIES ${BROTLIDEC_LIBRARIES} ${BROTLICOMMON_LIBRARIES})
     endif()
 endif()
+
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.5.1)
```
Comment (dg0yt) [inline ports/freetype/vcpkg-cmake-wrapper.cmake:72]: As above.

File: ports/freetype/vcpkg-cmake-wrapper.cmake [hunk 6]
```diff
@@ -63,3 +68,7 @@ if(FREETYPE_LIBRARIES)
         list(APPEND FREETYPE_LIBRARIES ${BROTLIDEC_LIBRARIES} ${BROTLICOMMON_LIBRARIES})
     endif()
 endif()
+
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.5.1)   
+    cmake_policy(POP)
+endif()
```
Comment (dg0yt) [inline ports/freetype/vcpkg-cmake-wrapper.cmake:74]: Please mind the final line break.

**Review by dg0yt — COMMENTED** [review]
This whole PR is only needed to support projects which have a `cmake_minimum_required(VERSION ...) of less than 2.8.

**Comment by JackBoosY** [comment]
gstreamer regressions are a baseline issue.

**Pull request closed without merging**